### PR TITLE
Begin moving tests to their applicable packages

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Moved various test files out of the azblob package and into their
respective client packages.
Added internal/testcommon package for shared test helpers.
No functional or test changes.

- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
Dependent on https://github.com/Azure/azure-sdk-for-go/pull/18905

File: sdk/storage/azblob/appendblob/client_test.go [hunk 1]
```diff
@@ -0,0 +1,1443 @@
...
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/bloberror"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/container"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/internal/testcommon"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+func Test(t *testing.T) {
+	suite.Run(t, &AppendBlobRecordedTestsSuite{})
+	//suite.Run(t, &AppendBlobUnrecordedTestsSuite{})
```
Comment (RickWinter) [inline sdk/storage/azblob/appendblob/client_test.go:29]: Should this be uncommented?

Comment (jhendrixMSFT) [reply sdk/storage/azblob/appendblob/client_test.go:29]: No.  These tests don't have recordings at present.

File: sdk/storage/azblob/internal/testcommon/cpk.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	EncryptionKeySHA256: &testInvalidEncryptedHash,
+	EncryptionAlgorithm: &testEncryptionAlgorithm,
+}
+
+var testEncryptedScope = "blobgokeytestscope"
+var TestCPKByScope = blob.CpkScopeInfo{
+	EncryptionScope: &testEncryptedScope,
+}
+
+var testInvalidEncryptedScope = "mumbojumbo"
```
Comment (RickWinter) [inline sdk/storage/azblob/internal/testcommon/cpk.go:34]: Make this a different value from the testInvalidEncryptedKey 
```suggestion
var testInvalidEncryptedScope = "mumbojumboscope"
```

**Review by RickWinter — COMMENTED** [review]
Is this a step 1 that will be followed up with work to fix the commented tests?

**Comment by jhendrixMSFT** [comment]
Commented tests will be handled separately and is tracked by https://github.com/Azure/azure-sdk-for-go/issues/18933.

The current goal is to move all package-specific tests to their respective sub-packages.

File: sdk/storage/azblob/internal/testcommon/cpk.go (hunk 2 above)
Comment (jhendrixMSFT) [reply sdk/storage/azblob/internal/testcommon/cpk.go:34]: I can clean this up, however the overall goal in this PR is to not make any functional test changes (so they don't get lost in the refactoring).

File: sdk/storage/azblob/appendblob/client_test.go (hunk 1 above)
Comment (jhendrixMSFT) [reply sdk/storage/azblob/appendblob/client_test.go:29]: @mohsha-msft is it just that these tests are missing recordings, or is there more to it than that.

Comment (jhendrixMSFT) [reply sdk/storage/azblob/appendblob/client_test.go:29]: Opened https://github.com/Azure/azure-sdk-for-go/issues/18935 to track this, we don't have to solve it here.

**Review by richardpark-msft — APPROVED** [review]
Good with it since it's an intermediate review, although @RickWinter has pointed out some things that might warrant us taking a more detailed look once you start checking them in "for real".

**Pull request merged**